### DISCUSSION TITLE
Observability library for Akka

### DISCUSSION THREAD

**Discussion by jacum** [apache/pekko#64] [Ideas]
Ladies and gentlemen, 

I am the author and maintainer of Akka Sensors (https://github.com/jacum/akka-sensors) library, native Prometheus monitors for Akka and around. It is an alternative to Lightbend Cinnamon and open-source Kamon and currently used in production at the Dutch ING bank and OnAir Entertainment. Not a 100% Cinnamon drop-in replacement (cluster features aren't covered) - but gives quite useful insights nevertheless.

Would it make sense to include it as a standard part for Pekko, to give potential Akka users switching to Pekko at least something, instead of Cinnamon (part of paid Lightbend license, not available for Pekko users). It also opens a way to further development of built-in instrumentations, e.g. cluster features.

**Comment by pjfanning** [comment]
@jacum thanks for your interest. We are being very conservative about what goes into Apache Pekko v1.0.0. Basically, it is based on the most recent Apache licensed Akka releases. Nothing else gets added. We are just changing package names, jar names, docs, etc to reflect the move to ASF.

For a v1.1.0 release, we will consider extra changes. This subsequent release feels like it won't happen for quite some time based on current v1.0.0 velocity.

There are a lot of modules that Pekko team need to maintain. Is there a reason, why a 'Pekko Sensors' could not remain a separate project - at least for the time being?

It is just my opinion, but I would prefer to make changes to Pekko core that make external ecosystem libs easier to write and maintain - as opposed to bringing them all in-house. 

I have https://github.com/pjfanning/micrometer-akka and I'm considering creating a pekko version but not to bring it into Pekko. But if 'Pekko Sensors' were to look like a better bet, I might abandon 'Micrometer-Pekko'.

Does this make sense?

**Reply by Roiocam** [reply]
> I have https://github.com/pjfanning/micrometer-akka and I'm considering creating a pekko version but not to bring it into Pekko. But if 'Pekko Sensors' were to look like a better bet, I might abandon 'Micrometer-Pekko'.

this repo seems like a base on Kamon, for me, Kamon is not quite useful. we are a custom own agent based on OpenTelemetry, we believe it is the Future standard of Observability. (In the same way, using bytebuddy and java instrumentation API is another option, but there is too much work to do)

for now, it supports:

- akka-persistence
- akka-projection
- akka-persistence-jdbc

**Reply by jphelp32** [reply]
@pjfanning have there been any decisions made re: the future of micrometer-akka/pekko? and/or this Sensors library?
thanks

**Reply by pjfanning** [reply]
The Pekko team have enough on their plates. There are no plans to take over extra projects. Micrometer-Pekko is my personal project and there are snapshots published. I intend to do a release after pekko core libs are released.

**Reply by pjfanning** [reply]
If other lib maintainers want to do pekko friendly libraries - the snapshots pekko jars are all there.

**Reply by jphelp32** [reply]
Ah, I had not seen your [micrometer-pekko](https://github.com/pjfanning/micrometer-pekko) repo. Looks good. thx

**Reply by jacum** [reply]
Just in case, my Sensors port:
https://github.com/jacum/pekko-sensors
